Detections/AuditLogs/UserAddedtoAdminRole.yaml

```diff
@@ -20,11 +20,11 @@ tags:
   - AADSecOpsGuide
 query: |
   AuditLogs
-    | where OperationName in ("Add eligible member (permanent)", "Add eligible member (eligible)")
-    | extend Role = tostring(TargetResources[0].displayName)
+    | where OperationName in ("Add eligible member (permanent)", "Add eligible member (eligible)", "Add member to role")
+    | extend Role = iff(OperationName=="Add member to role", tostring(parse_json(tostring(parse_json(tostring(TargetResources[0].modifiedProperties))[1].newValue))),tostring(TargetResources[0].displayName))
     | where Role contains "admin"
-    | extend AddedBy = tostring(parse_json(tostring(InitiatedBy.user)).userPrincipalName)
-    | extend AddedUser = tostring(TargetResources[2].userPrincipalName)
+    | extend AddedBy = iff(tostring(parse_json(tostring(InitiatedBy.user)).userPrincipalName)=="", tostring(parse_json(tostring(InitiatedBy.app)).displayName), tostring(parse_json(tostring(InitiatedBy.user)).userPrincipalName))
```

Comment:
InitiatedBy.user.userPrincipalName and InitiatedBy.app.displayName will get you the required data so no need to parse_json for it.

Reply:
Great suggestion, changed :)